src/node_crypto.cc

```diff
@@ -5937,6 +5937,33 @@ void SetEngine(const FunctionCallbackInfo<Value>& args) {
   if (r == 0)
     return ThrowCryptoError(env, ERR_get_error());
 }
+
+
+void GetEngines(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  ENGINE* e;
+  int i = 0;
+  Local<Array> arr = Array::New(env->isolate());
+
+  for (e = ENGINE_get_first(); e != nullptr; e = ENGINE_get_next(e)) {
+    const char* id = ENGINE_get_id(e);
+    const char* name = ENGINE_get_name(e);
```

Comment:
Good catch. `SetEngine` did not have `ENGINE_set_flag` so that flags was always 0 for dynamic engines. Fixed and add a new flags property.